### PR TITLE
SDI-538 create adjustment happy path

### PR DESCRIPTION
**Pull request by andymarke**
Test for happy path 
TODO
* Exception handling
* Retry

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -24,6 +24,8 @@ dependencies {
   implementation("org.springdoc:springdoc-openapi-starter-webmvc-ui:2.0.2")
   implementation("com.fasterxml.jackson.module:jackson-module-kotlin:2.14.1")
   implementation("org.jetbrains.kotlinx:kotlinx-coroutines-reactor")
+  implementation("org.jetbrains.kotlinx:kotlinx-coroutines-jdk8")
+
```
Comment (andymarke) [inline build.gradle.kts:28]: to convert back to Completable Future

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsService.kt [hunk 2]
```diff
@@ -1,13 +1,87 @@
...
+            ),
+            null
+          )
+        }
+      }
+    }
+  }
 }
+
+data class AdditionalInformation(
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsService.kt:80]: guesses on the message

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/integration/SqsIntegrationTestBase.kt [hunk 3]
```diff
@@ -35,8 +36,10 @@ import uk.gov.justice.hmpps.sqs.HmppsTopic
 )
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 @ActiveProfiles("test")
-@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_CLASS)
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/integration/SqsIntegrationTestBase.kt:38]: doesn't look like this was need and just slows tests

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsToNomisTest.kt [hunk 4]
```diff
@@ -0,0 +1,398 @@
...
+        @Test
+        fun `will create success telemetry`() {
+          await untilAsserted {
+            verify(telemetryClient).trackEvent(
+              eq("sentencing-adjustment-create-success"),
+              org.mockito.kotlin.check {
+                assertThat(it["sentenceAdjustmentId"]).isEqualTo(ADJUSTMENT_ID)
+                assertThat(it["nomisAdjustmentId"]).isEqualTo(nomisAdjustmentId.toString())
+                assertThat(it["offenderNo"]).isEqualTo(OFFENDER_NUMBER)
+              },
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsToNomisTest.kt:112]: should probably log booking id as well,  next PR

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsService.kt [hunk 5]
```diff
@@ -1,13 +1,87 @@
...
 @Service
 class SentencingAdjustmentsService(
   private val sentencingAdjustmentsApiService: SentencingAdjustmentsApiService,
   private val nomisApiService: NomisApiService,
   private val sentencingAdjustmentsMappingService: SentencingAdjustmentsMappingService,
+  private val telemetryClient: TelemetryClient
 ) {
-  // TODO: all sorts of stuff
+  suspend fun createAdjustment(createEvent: AdjustmentCreatedEvent) {
+    sentencingAdjustmentsMappingService.getMappingGivenSentenceAdjustmentId(createEvent.additionalInformation.id)
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsService.kt:16]: minor but should getMappingGivenSentenceAdjustmentId be the more generic getMappingGivenSentencingAdjustmentId

**Review by lbennett-moj — APPROVED** [review]

**Pull request merged**